app/components/ui/atoms/CalloutBox.js

```diff
@@ -2,16 +2,36 @@ import React from 'react'
 import styled from 'styled-components'
 import { Box } from 'grid-styled'
 import { th } from '@pubsweet/ui-toolkit'
+import Icon from './Icon'
 
-const CalloutBox = styled(({ enabled, ...rest }) => <Box {...rest} />).attrs({
-  mx: -3,
-  px: 3,
-  mb: 3,
+const Frame = styled(props => <Box {...props} />).attrs({
+  mb: 2,
 })`
   border: ${th('borderWidth')} ${th('borderStyle')} ${th('borderColor')};
   border-radius: ${th('borderRadius')};
-  border-color: ${({ enabled = true }) =>
-    enabled ? th('colorBorder') : 'transparent'};
+  border-color: ${th('colorBorder')};
+  position: relative;
 `
 
+const CloseButton = styled(Icon)`
+  position: absolute;
+  top: 0px;
+  right: 0px;
+  padding: ${th('space.2')};
+  svg {
+    stroke: ${th('colorFurniture')};
+    fill: ${th('colorFurniture')};
+  }
+`
+
+const CalloutBox = ({ closable, onClose, children, ...props }) => (
+  <Frame {...props}>
+    {closable && (
+      <CloseButton onClick={onClose} size={3}>
```

Comment:
An X in the top right should always be for closing shouldn't it? I feel that *how* the closing is achieved is a concern of the component above this one though.

Reply:
But you can't _mandate_ that the component above attaches a handler which applies closing functionality.

I could use `CalloutBox` elsewhere and pass in my own `handleClick` to the `onClose` prop, which does something totally different from "closing" e.g. turns the colour of some text red:

```
class myParentComponent extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      textColour: 'black'
    }
  }
  
  handleClick = () => {
    this.setState({textColor: 'red})
  }

  render() {
    <div>
      <h2 color={this.state.textColor}>Text here</h2>
    <CalloutBox
        closable
        onClose={handleClick}
    >
  }
}
```

Just because you call the prop `onClose`, doesn't mean that's what clicking an X actually does. A more accurate name for this prop would be `onXIconClick`.

This is the inherent problem with separating which component renders the X and which component handles the clicking of the X.

Do you see what I mean?

Reply:
You can't stop people from misusing your component but making your app do anything other than close as a result of someone clicking an X icon in the top right of a box is silly. 

I think perhaps the issue is that "close" isn't the event that has taken place. So in that sense `onXIconClick` is more accurate but accuracy isn't everything and to me at least `onClose` is more helpful.

I would settle for `onCloseClick` if pushed.

Reply:
I see what you mean but surely by the same argument Formik's `onSubmit` should be called `onSubmitButtonClick`?

Reply:
Ahh yeah, I see what you mean about misusing a component :thinking:

Perhaps my issue is more that this abstraction seems kinda arbitrary. We don't need a boxed component with an X for anything other than the situation where it's combined with a link that generates the box(es) in the first place.

So I don't see any need to remove the existence of the X icon from its intended purpose & it kinda makes the `FormSections.js` code harder to reason about. All for the sake of re-using a border in 2 places. Just feels like the wrong abstraction to me :woman_shrugging: 

It would make more sense to me, if we were building molecules that combined components for encapsulated behaviour i.e. if we built 2 component based on different ways to hide/show:
- checkbox generated boxes - checking generates & closes
- link generated/X closed boxes

The fact that both of these create a border around some fields doesn't feel like an important thing to group together?

And if we grouped the components this way instead, it would make for a page whose render function was more explicit and therefore easier to reason about.